clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1154,3 +1156,70 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
   }
   return false;
 }
+
+static bool calculateIsIntangibleType(QualType Ty) {
+  Ty = Ty->getCanonicalTypeUnqualified();
+  if (Ty->isBuiltinType())
+    return Ty->isHLSLSpecificType();
+
+  llvm::SmallVector<QualType, 8> TypesToScan;
```

Comment:
Do you have a reason to expect 8 types specifically? If not, it's better to use `llvm::SmallVector<QualType>`, which will calculate a reasonable small size based on the size of the elements.

Reply:
Thanks! For some reason I thought the capacity is a required parameter.